lib/browser/api/menu.js

```diff
@@ -175,6 +172,16 @@ Menu.buildFromTemplate = function (template) {
 
 /* Helper Functions */
 
+// validate the template against having the wrong attribute
+function typeCheckTemplate (template) {
```

Comment:
extremely minor style nit: it would be a little nicer to define this function as `isValidTemplate` and have it return `true`/`false`, and have the caller be responsible for deciding what to do with it (in this case, throw an exception). The reasoning is:
- it's easier for someone reading the code that calls this function to know what's going on. It's easy to guess that `isValidTemplate(template)` is going to do some sort of checking and return a boolean. `typeCheckTemplate(template)` is a little more obscure—what's it going to do if the check fails? It's unclear when reading that call what the outcome will be.
- it's a little more versatile in its usage, because you can combine it with other conditions or recover in different ways in different situations.

In this case it's an extremely minor point because the function is pretty simple and is only called in one place with one very specific purpose, so I wouldn't hold up the PR for it. But I figured it might be helpful to you to hear what my thoughts were on it :)